sklearn/pipeline.py

```diff
@@ -667,6 +667,36 @@ def _pairwise(self):
         # check if first estimator expects pairwise input
         return getattr(self.steps[0][1], "_pairwise", False)
 
+    def get_feature_names_out(self, input_features=None):
+        """Get output feature names for transformation.
+
+        Transform input features using the pipeline.
+        If the last step is a transformer, it's included
+        in the transformation, otherwise it's not.
+
+        Parameters
+        ----------
+        input_features : array-like of str or None, default=None
+            Input features.
+
+        Returns
+        -------
+        feature_names : array-like of string
+            Transformed feature names
+        """
+        feature_names = input_features
+        for _, name, transform in self._iter():
+            if not hasattr(transform, "get_feature_names_out"):
+                raise AttributeError(
+                    "Estimator {} does not provide get_feature_names_out. "
+                    "Did you mean to call Pipeline[:-1].get_feature_names_out"
```

Comment:
Maybe even:

```suggestion
                    "Did you mean to call pipeline[:-1].get_feature_names_out"
```

since `Pipeline` can be interpreted as the non instantiated class which is not sliceable.